ports/nrf/common-hal/bleio/Adapter.c

```diff
@@ -39,17 +39,34 @@
 #include "supervisor/usb.h"
 #include "shared-bindings/bleio/Adapter.h"
 #include "shared-bindings/bleio/Address.h"
+#include "mpconfigboard.h" // for BOARD_HAS_CRYSTAL
 
 STATIC void softdevice_assert_handler(uint32_t id, uint32_t pc, uint32_t info) {
     mp_raise_msg_varg(&mp_type_AssertionError,
                       translate("Soft device assert, id: 0x%08lX, pc: 0x%08lX"), id, pc);
 }
 
+static inline bool board_has_crystal(void) {
+#ifdef BOARD_HAS_CRYSTAL
+    return BOARD_HAS_CRYSTAL == 1;
+#else
+    return false;
+#endif
+}
+
 STATIC uint32_t ble_stack_enable(void) {
-    nrf_clock_lf_cfg_t clock_config = {
+    nrf_clock_lf_cfg_t clock_config;
+    if (board_has_crystal()) {
+        clock_config = (nrf_clock_lf_cfg_t){
         .source = NRF_CLOCK_LF_SRC_XTAL,
-        .accuracy = NRF_CLOCK_LF_ACCURACY_20_PPM
-    };
+        .accuracy = NRF_CLOCK_LF_ACCURACY_20_PPM};
+    } else {
+        clock_config = (nrf_clock_lf_cfg_t){
```

Comment:
Indentation seems slightly unusual here, consider reindenting.

Reply:
Could you please elaborate.
It seems okay in my editor.

Reply:
In the view here in github, ".source" is directly below "clock_config" on the previous line, I would expect it to be further indented.  (however, in *this* view, the line I'm talking about is cut off)

Reply:
Now i understand what you meant and i fixed it,
Does it look good now?